File: intro-to-computer-science/week-1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>


void print_row( int spaces, int hashes);

int main(void)
{
  int height;
  do
  {
      height = get_int("Height must be between 1 and 8, inclusive, please try again: ");
  }
  while (height < 1 || height > 8);

  // call the print_row function for the number of times of th user's input
  for (int i = 1; i <= height; i++)
  // Logic:  This prints our each row of the pyramid.
  {
    print_row(height - i, i);
  }
}


void print_row(int spaces, int hashes)
{
  // Delcare the characters to be printed
  char space = ' ';
  char hash = '#';
  char *gap = "  ";

  // Space Printing
  for (int j = 0; j < spaces; j++)
  // Logic: iterate the number of times where spaces is > 0 but < height
  // Because the last row will have 0 space and all hashes., so we EXCLUDE the height.
  {
    printf("%c", space);
  }

  // Hash Printing
  for (int j = 0; j < hashes; j++)
  // Lofic: iterate the number of times where hashes is > 0 but <= height
  // Because the last row will have all hashes and no spaces, so we INCLUDE the height.
  {
    printf("%c", hash);
  }

  // Fixed Space Printing
  printf("%s", gap);

  // Hash Printing
  for (int j = 0; j < hashes; j++)
  // Lofic: iterate the number of times where hashes is > 0 but <= height
  // Because the last row will have all hashes and no spaces, so we INCLUDE the height.
  {
    printf("%c", hash);
  }

  // print a new line after each row
  printf("\n");
}

File: intro-to-computer-science/week-6/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    # get the user's input for number of blocks
    height = 0
    min_height = 1
    max_height = 8

    height = get_int("Height: ")

    while height < min_height or height > max_height:
        height = get_int(f"Height must be between {min_height} - {max_height}: ")

    # Iterate throught the blocks to print in reverse with space
    for i in range(1, height + 1, 1):
        print_row(spaces=height - i, hashes=i)


def print_row(spaces: int, hashes: int):
    print(" " * spaces + "#" * hashes)


if __name__ == "__main__":
    main()
